addons/mail/wizard/mail_template_preview.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+LIST_FIELDS_MAIL_TEMPLATE = ['subject', 'body_html', 'email_from', 'email_to', 'partner_to', 'email_cc', 'reply_to', 'scheduled_date', 'attachment_ids']
+
+
+class MailTemplatePreview(models.TransientModel):
+    _inherit = "mail.template"
+    _name = "mail.template.preview"
+    _description = "Email Template Preview"
+
+    @api.model
+    def _selection_target_model(self):
+        return [(model.model, model.name) for model in self.env['ir.model'].search([])]
+
+    @api.model
+    def _selection_languages(self):
+        return self.env['res.lang'].get_installed()
+
+    @api.model
+    def default_get(self, fields):
+        result = super(MailTemplatePreview, self).default_get(fields)
+        default_mail_template_id = self.env.context.get('default_mail_template_id')
+        if not default_mail_template_id or 'resource_ref' not in fields:
+            return result
+        mail_template = self.env['mail.template'].browse(default_mail_template_id)
+        res = self.env[mail_template.model_id.model].search([], limit=1)
+        if res:
+            result['resource_ref'] = '%s,%s' % (mail_template.model_id.model, res.id)
+        return result
+
+    mail_template_id = fields.Many2one('mail.template', string="Related Mail Template")
+    model_id = fields.Many2one('ir.model', string='Targeted model', related="mail_template_id.model_id")
+    resource_ref = fields.Reference(string='Record reference', selection='_selection_target_model')
+    lang = fields.Selection(_selection_languages, string='Template Preview Language')
+    partner_ids = fields.Many2many('res.partner', string='Recipients')
+    # Override fields from mail.template model
+    attachment_ids = fields.Many2many(string='Attachments', store=False)
+    # Error Helper fields
+    error_msg = fields.Char('Error Message', readonly=True)
+
+    @api.onchange('lang', 'resource_ref')
+    def on_change_resource_ref(self):
+        mail_template = self.mail_template_id.with_context(lang=self.lang)
```

Comment:
You copy/pasted my code and I did not include your docstring (oops)

Hopefully, I saved it somewhere in case this happened ^^

```
""" Preview the mail template (body, subject, ...) depending of the language and
the record reference, more precisely the record id for the defined model of the mail template.
If no record id is selectable/set, the jinja placeholders won't be replace in the display information. """
```

Reply:
Oops, I didn't notice that. :+1: